### PR TITLE
Add getDefaultCard prop to editor

### PR DESCRIPTION
**Pull request by JoelArmendariz**
Closes #

**Summary**
Need to give the consumer the ability to create their own default card configs.

**Change List (commits, features, bugs, etc)**
Add `getDefaultCard` prop to editor

**Acceptance Test (how to verify the PR)**
Go to the `With get default card` story and see that the card properly renders a custom default config

**Regression Test (how to make sure this PR doesn't break old functionality)**
Make sure that the card configs (found in the json editor modal) are still the correct defaults for mainstream cards.

### CODE REVIEW

File: packages/react/src/components/DashboardEditor/DashboardEditor.jsx [hunk 1]
```diff
@@ -459,11 +465,16 @@ const DashboardEditor = ({
...
-      // notify consumers that the card has been added if they're listening (they might want to tweak the card defaults)
-      const cardConfig = onCardChange
-        ? onCardChange(getDefaultCard(type, mergedI18n), dashboardJson)
+      const defaultCard = customGetDefaultCard // Use the default card specified by the consumer if it exists
+        ? customGetDefaultCard(type)
         : getDefaultCard(type, mergedI18n);
 
+      // notify consumers that the card has been added if they're listening (they might want to tweak the card defaults here)
+      const cardConfig =
+        onCardChange && !customGetDefaultCard
```
Comment (scottdickerson) [inline packages/react/src/components/DashboardEditor/DashboardEditor.jsx:474]: is this right?  Should we call onCardChange even if they have a customGetDefaultCard or is the comment above the line incorrect?  Seems like they disagree.

**Review by scottdickerson — APPROVED** [review]
one small question but otherwise looks good

File: packages/react/src/components/DashboardEditor/DashboardEditor.jsx (hunk 1 above)
Comment (JoelArmendariz) [reply packages/react/src/components/DashboardEditor/DashboardEditor.jsx:474]: Right, the comment is left over and incorrect. To me it makes sense to call onCardChange when adding a new card IF there is no getDefaultCard passed, but if there is one passed, I would think the consumer would want to handle it there instead. I'll remove the misleading comment.

**Review by kevinsperrine — APPROVED** [review]

File: packages/react/src/components/DashboardEditor/DashboardEditor.jsx (hunk 1 above)
Comment (davidicus) [reply packages/react/src/components/DashboardEditor/DashboardEditor.jsx:474]: Can we fix the comment?

**Review by davidicus — APPROVED** [review]

**Pull request merged**